OPHD/States/MapViewState.cpp

```diff
@@ -1412,18 +1412,24 @@ void MapViewState::updateCommRangeOverlay()
 
 void MapViewState::updatePoliceOverlay()
 {
-	resetPoliceOverlays();
+	if (mTileMap != nullptr)
+	{
+		resetPoliceOverlays();
 
-	auto& structureManager = NAS2D::Utility<StructureManager>::get();
-	fillOverlay(*mTileMap, mPoliceOverlays[0], structureManager.getStructures<SurfacePolice>());
-	fillOverlay(*mTileMap, mPoliceOverlays, structureManager.getStructures<UndergroundPolice>());
+		auto& structureManager = NAS2D::Utility<StructureManager>::get();
+		fillOverlay(*mTileMap, mPoliceOverlays[0], structureManager.getStructures<SurfacePolice>());
+		fillOverlay(*mTileMap, mPoliceOverlays, structureManager.getStructures<UndergroundPolice>());
+	}
 }
 
 
 void MapViewState::resetPoliceOverlays()
 {
-	const auto adjustedZ = mTileMap->maxDepth() + 1;
-	mPoliceOverlays = std::vector<std::vector<Tile*>>(static_cast<std::size_t>(adjustedZ));
+	if (mTileMap != nullptr)
+	{
+		const auto adjustedZ = mTileMap->maxDepth() + 1;
+		mPoliceOverlays = std::vector<std::vector<Tile*>>(static_cast<std::size_t>(adjustedZ));
+	}
```

Comment:
It might be preferable to use an early `return`, rather than indent all the mainline code.

----

Perhaps a bigger question is how or why is `mTileMap` not initialized before this is called. It would seem there may be a structural defect with the code, which is being compensated for here, rather than fixed at the source. This check seems to imply bad design on our part somewhere. Perhaps we should identify and fix the design that makes that check necessary.